crates/client/src/storage/fs.rs

```diff
@@ -328,6 +435,49 @@ impl ContentStorage for FileSystemContentStorage {
     }
 }
 
+/// Represents a namespace_domain map storage using the local file system.
+pub struct FileSystemNamespaceMapStorage {
+    base_dir: PathBuf,
+}
+
+impl FileSystemNamespaceMapStorage {
+    /// Creates new namespace_domain mapping config
+    pub fn new(base_dir: impl Into<PathBuf>) -> Self {
+        Self {
+            base_dir: base_dir.into(),
+        }
+    }
+}
+
+#[async_trait]
+impl NamespaceMapStorage for FileSystemNamespaceMapStorage {
+    async fn load_namespace_map(&self) -> Result<Option<HashMap<String, String>>> {
+        let namespace_path = &self.base_dir;
+        let namespace_map = load(namespace_path).await?;
+        Ok(namespace_map)
+    }
+
+    async fn reset_namespaces(&self) -> Result<()> {
+        remove(&self.base_dir).await?;
+        Ok(())
+    }
+
+    async fn store_namespace(&self, namespace: String, registry_domain: String) -> Result<()> {
+        let mapping = self.load_namespace_map().await?;
+        if let Some(mut mapping) = mapping {
+            mapping.insert(namespace, registry_domain);
+            let json = serde_json::to_string(&mapping)?;
+            fs::write(&self.base_dir, json)?;
+        } else {
+            let mut mapping = HashMap::new();
+            mapping.insert(namespace, registry_domain);
+            let json = serde_json::to_string(&mapping)?;
+            fs::write(&self.base_dir, json)?;
+        }
```

Comment:
I think this should work:

```suggestion
        let mut mapping = self.load_namespace_map()
            .await?
            .unwrap_or_default();
        mapping.insert(namespace, registry_domain);
        let json = serde_json::to_string(&mapping)?;
        fs::write(&self.base_dir, json)?;
```